corehq/apps/users/user_data.py

```diff
@@ -63,10 +64,31 @@ def _provided_by_system(self):
 
     def to_dict(self):
         return {
+            **self._schema_defaults,
             **self._local_to_user,
             **self._provided_by_system,
         }
 
+    @property
+    def _schema_defaults(self):
+        fields = self._schema_fields
+        return {field.slug: '' for field in fields}
+
+    @cached_property
+    def _schema_fields(self):
+        from corehq.apps.users.views.mobile.custom_data_fields import (
+            CUSTOM_USER_DATA_FIELD_TYPE,
+        )
+
+        try:
+            definition = CustomDataFieldsDefinition.objects.get(
+                domain=self.domain, field_type=CUSTOM_USER_DATA_FIELD_TYPE)
```

Comment:
Looks like this introduces a DB dependency that some tests don't like.  You might consider extending this test decorator to address that:
https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/tests/util.py#L23-L24